src/shared/filesystemUtilities.ts

```diff
@@ -65,14 +70,16 @@ export async function findFileInParentPaths(searchFolder: string, fileToFind: st
     return findFileInParentPaths(parentPath, fileToFind)
 }
 
-export const getTempDirPath = (prefix: string = 'vsctk') => {
-    return path.join(
-        os.type() === 'Darwin' ? '/tmp' : os.tmpdir(),
-        prefix
-    )
-}
-
 const _mkdtemp = promisify(fs.mkdtemp)
-export const  mkdtemp = async (prefix?: string) => {
-    return _mkdtemp(getTempDirPath(prefix))
+export const  mkdtemp = async (...relativePathParts: string[]) => {
+    const _relativePathParts = relativePathParts || ['vsctk']
+    const tmpPath = path.join(tempDirPath, ..._relativePathParts)
+    const tmpPathParent = path.dirname(tmpPath)
+    // fs.mkdtemp fails on OSX if prefix contains path separator
+    // so we must create intermediate dirs if needed
+    if (!await fileExists(tmpPathParent)) {
+        await mkdir(tmpPathParent)
```

Comment:
`fs.mkdir` is [not recursive by default](https://nodejs.org/api/fs.html#fs_fspromises_mkdir_path_options), so you'll need to do:

```typescript
await mkdir(tmpPathParent, { recursive: true })
```

Reply:
✔️